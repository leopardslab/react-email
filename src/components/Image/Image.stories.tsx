import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Image } from './Image';

export default {
  component: Image,
} as ComponentMeta<typeof Image>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Image> = (args) => (
  <table>
    <tbody>
      <tr>
        <Image {...args} />
      </tr>
    </tbody>
  </table>
);

export const Default = Template.bind({});

Default.args = {
  src: 'https://images.unsplash.com/photo-1453728013993-6d66e9c9123a',
  alt: 'This is a random image',
  width: '250px',
  style: {
    borderRadius: '10px',
  },
};
