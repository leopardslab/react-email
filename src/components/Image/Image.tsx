import { CSSProperties } from 'react';
import { makeStyles } from '../../utils/makeStyles';
import { sx } from '../../utils/sx';
import { BaseStyleProp } from '../types';

type ImageStyles =
  | 'root'
  | 'table'
  | 'body'
  | 'image'
  | 'caption'
  | 'imageSection'
  | 'captionSection'
  | 'imageColumn';

export interface ImageProps extends BaseStyleProp<ImageStyles> {
  src: string;
  alt: string;
  caption?: string;
  width: CSSProperties['width'];
  height?: CSSProperties['height'];
  captionAlign?: 'left' | 'center' | 'right';
}

const useStyles = makeStyles({
  root: { border: '0', display: 'block', outline: 'none' },
  table: { width: 'fit-content', border: '0', verticalAlign: 'top' },
  body: {},
  image: {},
  caption: {},
  imageSection: {},
  captionSection: {},
  imageColumn: {},
});

export const Image = ({
  src,
  alt,
  caption,
  width,
  height = 'auto',
  classes,
  className,
  captionAlign = 'center',
}: ImageProps): JSX.Element => {
  const styles = useStyles({ classes });

  return caption ? (
    <table cellPadding="0" cellSpacing="0" role="presentation" style={styles.table}>
      <tbody style={styles.body}>
        <tr style={styles.imageSection}>
          <td style={styles.imageColumn}>
            <img
              className={className}
              height={height}
              width={width}
              src={src}
              alt={alt}
              style={sx(styles.root, styles.image)}
            />
          </td>
        </tr>
        <tr style={styles.captionSection}>
          <td style={sx(styles.caption, { textAlign: captionAlign })} id="imageCaption">
            {caption}
          </td>
        </tr>
      </tbody>
    </table>
  ) : (
    <img
      className={className}
      height={height}
      width={width}
      src={src}
      alt={alt}
      style={sx(styles.root, styles.image)}
    />
  );
};
