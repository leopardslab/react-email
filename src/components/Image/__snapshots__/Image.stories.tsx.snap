// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Image Default smoke-test 1`] = `
<div style="margin: 0px auto; max-width: 600px;">
  <table cellpadding="0"
         cellspacing="0"
         role="presentation"
         style="border: 0px; vertical-align: top; width: 100%;"
  >
    <tbody>
      <tr>
        <td align="center"
            style="vertical-align: top;"
        >
          <h2 style="margin: 0px; padding: 0px; text-align: center;">
            Hello World
          </h2>
        </td>
      </tr>
    </tbody>
  </table>
  <hr style="margin-left: 0px; border-top: 1px solid; width: 100%;">
  <table cellpadding="0"
         cellspacing="0"
         role="presentation"
         style="border: 0px; vertical-align: top;"
  >
    <tbody>
      <tr>
        <td align="center"
            style="vertical-align: top;"
        >
          <table cellpadding="0"
                 cellspacing="0"
                 role="presentation"
                 style="width: fit-content; border: 0px; vertical-align: top;"
          >
            <tbody>
              <tr>
                <td>
                  <img height="auto"
                       width="250px"
                       src="https://images.unsplash.com/photo-1453728013993-6d66e9c9123a"
                       alt="This is a random image"
                       style="border: 0px; display: block; outline: none; border-radius: 10px;"
                  >
                </td>
              </tr>
              <tr>
                <td id="imageCaption"
                    style="text-align: center;"
                >
                  This is a caption
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td align="center"
            style="padding-left: 20px; vertical-align: top;"
        >
          <h2 style="margin: 0px; padding: 0px; text-align: left;">
            This is a random text
          </h2>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
